checker-qual/src/main/java/org/checkerframework/checker/regex/qual/PolyRegex.java

```diff
@@ -11,10 +11,13 @@
 /**
  * A polymorphic qualifier for the Regex type system.
  *
- * <p>Any method written using {@link PolyRegex} conceptually has two versions: one in which every
- * instance of {@link PolyRegex} has been replaced by {@link Regex}, and one in which every instance
- * of {@link PolyRegex} has been replaced by {@link UnknownRegex}.
+ * <p>Any method written using {@link PolyRegex} can have the following possible values and {@link
+ * Regex} can take an integer argument for different instantiation.
  *
+ * <p>The possible value could be:
```

Comment:
Can you look how this appears in the generated Javadoc? I think it will be quite awkward to have the colon at the end of the paragraph and wonder where the list of elements is. And again, the "possible value" part does not make sense here, as you don't make the capturing group argument explicit.

Reply:
Also, there are `PartialRegex` and `RegexBottom` as other qualifiers in this hierarchy.

Reply:
add a test case